opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js

```diff
@@ -0,0 +1,42 @@
+import { elUpdateByQueryForMigration } from '../database/engine';
+import { READ_INDEX_INTERNAL_OBJECTS } from '../database/utils';
+import { DatabaseError } from '../config/errors';
+
+export const up = async (next) => {
+  const defaultDate = new Date();
+  defaultDate.setHours(0, 0, 0, 0);
+
+  await elUpdateByQueryForMigration(
+    '[MIGRATION] from Synchronizers current_state to current_state_date',
+    READ_INDEX_INTERNAL_OBJECTS,
+    {
+      script: {
+        source: 'if (ctx._source.current_state == params.emptyCurrentState) {'
+          + 'ctx._source.current_state_date = params.defaultDate'
+          + '} else {'
+          + 'ctx._source.current_state_date = ctx._source.current_state'
+          + '}'
+          + 'ctx._source.remove("current_state")',
+        params: {
+          emptyCurrentState: {},
```

Comment:
Should not be as string here? '{}'

Reply:
We tested it against problematic stored document and the migration executed flawlessly. What benefit do you see to make it a string?

Reply:
no it's an empty object, not a string